### ISSUE TITLE
Can (do?) we have a color detector?

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap--Build-Your-Own-Blocks#1878]
I'm wanting some way to read colors, like this:
![image](https://user-images.githubusercontent.com/6549787/31109100-b0a156d2-a7cb-11e7-925f-a48a0829bc33.png)
Is there one? Could there be?

Maybe at least this reporter could work in one of these predicates?
![image](https://user-images.githubusercontent.com/6549787/31109097-a34156d6-a7cb-11e7-82c3-6cd8d761c3c7.png)
![image](https://user-images.githubusercontent.com/6549787/31109140-fb59fcec-a7cb-11e7-954f-749b257a7c5d.png)

Relatedly, if I report PEN TRAILS, is there any way to process that data in Snap? Can I access the individual pixels somehow as if they were elements of a list?

**Comment by cycomachead** [comment]
Look in libraries for the media comp library -- that will solve some of the issues. 

We need first class colors for most of what you want, but I don't understand why Jens doesn't want them. 

We could make a "color I am touching", but it's a bit ambiguous how it might handle multiple possible colors at the same location (i.e. Overlapping sprites and stuff). Presumably it'd just be the color of the top layer at the center coordinates of a sprite. That seems doable with a Javascript function.

**Comment by brianharvey** [comment]
No, it shouldn't include sprite colors, imho.  Just pentrails and background layers.

Jens isn't against first class colors; he's just against doing it at the same time as the sprite OOP stuff.  It'll come.

What media comp library?  I see a pixels library in dev, but it doesn't seem to do what Mary wants.

**Comment by cycomachead** [comment]
A touching colors: only stage and pentrails makes sense, but then it makes it hard to detect what you’re eyes are seeing. 

As far as the libraries: my bad; I hadn’t closely looked at the pixels library. Based on Jens’ commit message I thought it included what was in the “media comp” project Jens built. I can share it later. :)


As far as timing: sure I guess, but we could have had first class colors over a year and a half ago and while they aren’t as computationally interesting as OOP, I think they solve more issues people are facing today. But anyway, as long as we get them, that’s good!
-- 
Michael Ball
From my iPhone
michaelball.co

**Comment by KingDavid12** [comment]
@maryfries I have a library you could try:
[ExtraPen.zip](https://github.com/jmoenig/Snap--Build-Your-Own-Blocks/files/1708121/ExtraPen.zip)

It adds blocks that allow you to mostly fake first class colors. It includes a "color I am touching" block.
